git-cliff-core/src/commit.rs

```diff
@@ -287,25 +294,17 @@ impl Commit<'_> {
 			if let (Some(field_name), Some(pattern_regex)) =
 				(parser.field.as_ref(), parser.pattern.as_ref())
 			{
-				regex_checks.push((
-					pattern_regex,
-					match field_name.as_str() {
-						"id" => Some(self.id.clone()),
-						"message" => Some(self.message.clone()),
-						"body" => body,
-						"author.name" => self.author.name.clone(),
-						"author.email" => self.author.email.clone(),
-						"committer.name" => self.committer.name.clone(),
-						"committer.email" => self.committer.email.clone(),
-						_ => None,
+				match tera::dotted_pointer(&lookup_context, field_name) {
```

Comment:
Hmm, this function is not documented anywhere. How did you find it? Do we know if it is stable?

Reply:
Found the functionality in the tera docs [here](https://keats.github.io/tera/docs/#dot-notation) which allows stuff like this to work: `{% if commit.github.username %} by @{{ commit.github.username }}{%- endif -%}`. Then just looked where user context is provided and picked around a little. Found in code [here](https://github.com/Keats/tera/blob/ae13d7ce39d732aae3f68435ed52c60732fe0ee0/src/context.rs#L244C1-L269C10).

Its true that this API is not documented, but its used by tera itself to make user contexts possible/easy to use. So IMHO if this is changed something similar has to be in tera itself to replace it. If this is of concern I can yeet the relevant code into a module and call it a day. The functionality is tested and benched.

Reply:
Got it, it makes sense to use it then. Good discovery 👍🏼